homeassistant/components/climate/fritzhome.py

```diff
@@ -0,0 +1,138 @@
+
+    OPERATION_LIST = [STATE_ECO, STATE_COMFORT]
+
+    def __init__(self, hass, device):
+        self._device = device
+        self._actual_temperature = None
+        self._target_temperature = None
+        self._eco_temperature = None
+        self._comfort_temperature = None
+
+    @property
+    def available(self):
+        """Return if thermostat is available."""
+        return self._device.present
+
+    @property
+    def name(self):
+        """Return the name of the device."""
+        return self._device.name
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement that is used."""
+        return TEMP_CELSIUS
+
+    @property
+    def precision(self):
+        """Return precision 0.5."""
+        return PRECISION_HALVES
+
+    @property
+    def current_temperature(self):
+        """Can not report temperature, so return target_temperature."""
+        return self._actual_temperature
+
+    @property
+    def target_temperature(self):
+        """Return the temperature we try to reach."""
+        return self._target_temperature
+
+    def set_temperature(self, **kwargs):
+        """Set new target temperature."""
+        temperature = kwargs.get(ATTR_TEMPERATURE)
+        if temperature is None:
+            return
+        self._device.set_target_temperature(temperature)
+
+    @property
+    def current_operation(self):
+        """Return the current operation mode."""
+        if not self.available:
+            return None
+        if self._target_temperature == self._comfort_temperature:
+            return STATE_COMFORT
+        elif self._target_temperature == self._eco_temperature:
+            return STATE_ECO
+        return 'unknown'
+
+    @property
+    def operation_list(self):
+        """Return the list of available operation modes."""
+        return self.OPERATION_LIST
+
+    @property
+    def min_temp(self):
+        """Return the minimum temperature."""
+        return self._eco_temperature
+
+    @property
+    def max_temp(self):
+        """Return the maximum temperature."""
+        return self._comfort_temperature
+
+    @property
+    def device_state_attributes(self):
+        """Return the device specific state attributes."""
+        attr = {
+            ATTR_AIN: self._device.ain,
+            ATTR_FW_VERSION: self._device.fw_version,
+            ATTR_ID: self._device.id,
+            ATTR_MANUFACTURER: self._device.manufacturer,
+            ATTR_PRODUCTNAME: self._device.productname,
+        }
+        return attr
+
+    def update(self):
+        """Update the data from the thermostat."""
+        try:
+            self._device.update()
+            self._actual_temperature = self._device.actual_temperature
+            self._target_temperature = self._device.target_temperature
+            self._comfort_temperature = self._device.comfort_temperature
+            self._eco_temperature = self._device.eco_temperature
+        except Exception as exc:
```

Comment:
I'm not sure if you shouldn't just keep the old values. Occasional update failures might happen, and setting everything to `None` could confuse the front end, automations etc.